pyrsistent/_pmap.py

```diff
@@ -140,7 +140,9 @@ def __str__(self):
 
     def __hash__(self):
         # This hashing algorithm is probably not the speediest
-        return hash(frozenset(self.iteritems()))
+        if not hasattr(self, '_cached_hash'):
```

Comment:
changed to `_pmap_hash`, hasattr is used elsewhere in pyrisistent, so it seems to be okay for this coding guide.